src/createReducer.js

```diff
@@ -361,6 +361,11 @@ function createReducer<M, L>(structure: Structure<M, L>) {
         result = setIn(result, 'values', values)
         result = setIn(result, 'initial', values)
       }
+      var submitFailed = getIn(state, 'submitFailed');
```

Comment:
Oh sorry my intention was to comment on the entire block. This seems to be checking if `submitFailed`, and if so, propagating `valid` state to the post-reset form state. My question is mainly, why would we want to propagate `invalid` after resetting?